README.md

```diff
@@ -111,6 +124,268 @@ instructions in its
 (coordinating guidance is in comments in the
 default recipe).
 
+
+## Other Guidelines
+
+### Modules and classes in cookbooks
+
+It is often useful to factor out logic into a library - especially logic that
+doesn't create resources. Doing so makes this logic easier to unit test and
+makes the recipe or resource cleaner.
+
+Our standard is that all cookbooks use the top-level container of `module FB`,
+and then create a class for their cookbook under that. For example, `fb_fstab`
+creates a `class Fstab` inside of the `module FB`. We will refer to this as
+the cookbook class from here.
+
+We require all cookbooks use this model for consitency.
+
+Since we don't put anything other than other classes inside the top-level
+object, it's clear that a `module` is the right choice.
+
+While there is no reason that a cookbook class can't be one design to be
+instantiated, more often than not it is simply a collection of callable class
+methods - a place to put methods that can then be called both from this
+cookbook and others.
+
+Below the cookbook class, the author is free to make whatever class or methods
+they desire.
+
+It is often desirable to `include` a module into a Custom Resource, and we do
+this by creating a module inside of the cookbook class, often ending in
+`Provider`, e.g. `FstabProvider`. This code should be in its own library file.
+
+It is worth noting that `fb_fstab` itself violates this guideline as it is
+quite old before we had solidified a standard and thus it's provider module
```

Comment:
s/it's/its/. Also maybe rewrite this whole sentence to read less like a stream of consciousness

Reply:
This paragraph is gone.